frontend/src/HomePage/SignHandlers/SignUp.jsx

```diff
@@ -19,8 +20,10 @@ function SignUp() {
             password: form.elements.password.value
         };
         try {
-            await submitProfile(userData);
-            navigate('/createprofile');
+            const response = await submitProfile(userData);
+            const data  = await response.json();
+            const id = data.userId
```

Comment:
Is it possible `.userId` property is null? Do we need to do a check first and error out if its not there?

Reply:
If userId was null that would suggest an error with signing since ids are automatically handled by the database, so this is implicitly being handled by the try-catch block that line is within

Reply:
From my personal experience, even if something should "technically never happen", with a large enough user base something is bound to go unexpected - so programming defensively is usually a good idea. If it'll be caught by the try catch then I guess that works.